src/Microsoft.OData.Core/Json/JsonValueUtils.cs

```diff
@@ -387,7 +389,13 @@ internal static void WriteEscapedJsonString(TextWriter writer, string inputStrin
                 for (; currentIndex < inputStringLength; currentIndex++)
                 {
                     char c = inputString[currentIndex];
-                    string escapedString = JsonValueUtils.SpecialCharToEscapedStringMap[c];
+                    string escapedString = null;
+
+                    if (stringEscapeOption == ODataStringEscapeOption.EscapeNonAscii ||
+                        (c < 0x7F && stringEscapeOption == ODataStringEscapeOption.EscapeOnlyControls))

```

Comment:
>&& stringEscapeOption == ODataStringEscapeOption.EscapeOnlyControls [](start = 34, length = 67)

Is this part of the condition required, or could we just check (stringEscapeOption == ODataStringEscapeOption.EscapeNonAscii || c < 0x7F)?

Reply:
Yes. Will change, thanks.